### PR TITLE
Add Firebase Analytics library into dependencies

### PR DESCRIPTION
**Pull request by binkos**
https://jira.epam.com/jira/browse/EPMEDU-753

Screen Shot 2022-10-12 at 12 44 30 PM

photo_2022-10-12_12-45-06

### CODE REVIEW

File: library/common/src/main/java/com/epmedu/animeal/common/di/AnalyticsModule.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.google.firebase.analytics.FirebaseAnalytics
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
```
Comment (egorikftp) [inline library/common/src/main/java/com/epmedu/animeal/common/di/AnalyticsModule.kt]: let's create separate library module `analytics`

Comment (binkos) [reply library/common/src/main/java/com/epmedu/animeal/common/di/AnalyticsModule.kt]: Yeah, I thought about it.
But from my point of view Analytics is common part of all screens and we will not win a lot from splitting it on separate modules. Isn't it?

Comment (binkos) [reply library/common/src/main/java/com/epmedu/animeal/common/di/AnalyticsModule.kt]: Resolved by reason: We try to remove all interactions with `common` module

File: library/analytics/src/main/java/com/epmedu/animeal/analytics/AnalyticsManagerImpl.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.epmedu.animeal.analytics
+
+import android.os.Bundle
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.google.firebase.analytics.ktx.logEvent
+import javax.inject.Inject
+
+class AnalyticsManagerImpl @Inject constructor(
```
Comment (egorikftp) [inline library/analytics/src/main/java/com/epmedu/animeal/analytics/AnalyticsManagerImpl.kt]: looks like can be `internal`

File: library/analytics/src/main/java/com/epmedu/animeal/analytics/di/AnalyticsModule.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Singleton
+    @Provides
+    fun providesAnalyticsManager(analytics: FirebaseAnalytics): AnalyticsManager =
+        AnalyticsManagerImpl(analytics)
+
+    @Singleton
+    @Provides
+    fun providesFirebaseAnalytics(@ApplicationContext context: Context): FirebaseAnalytics =
+        FirebaseAnalytics.getInstance(context)
```
Comment (egorikftp) [inline library/analytics/src/main/java/com/epmedu/animeal/analytics/di/AnalyticsModule.kt]: can be more simple with extension `Firebase.analytics`

File: settings.gradle.kts [hunk 4]
```diff
@@ -67,6 +67,7 @@ include(":feature:tabsflow:tabs")
 include(":feature:tabsflow:favourites")
 
 include(":library:common")
+include(":library:analytics")
```
Comment (egorikftp) [inline settings.gradle.kts:69]: can be above `include(":library:common")`

**Review by egorikftp — APPROVED** [review]

**Review by Fileek — APPROVED** [review]

**Pull request merged**